backend/fms_core/services/derived_sample.py

```diff
@@ -0,0 +1,43 @@
+from django.core.exceptions import ValidationError
+from fms_core.models import Biosample, DerivedSample, DerivedBySample, Sample, Container, Process
+
+def get_single_derived_sample_from_sample(sample):
+    derived_sample = None
+    errors = []
+    warnings = []
+
+    ds = DerivedSample.objects.filter(derived_by_sample__sample_id=sample.id)
+    if ds.count() > 1:
+        errors.append(f"")
+    else:
+        derived_sample = ds.first()
+
+    return (derived_sample, errors, warnings)
+
+
+def derive_sample_from_sample(sample, new_sample_data, new_derived_sample_data):
```

Comment:
This would need some work to be compatible with pools ... This will only be useful for extraction as far as i see. instead of getting a single derived_sample from the sample (which is not true in most cases) we could make it copy of all derived_sample and connect them to a new sample that is returned (we could have an array of derived_samples returned too). Also the name is confusing i think we should not use derive in it...

Reply:
Indeed this is not intended to handle the case of pools. We could handle this later on in the second part of the release since nowhere in the code the pools are handled for now.

Also, what name do you suggest for this function? Basically, the function creates a new derived_sample, a new sample, and the proper derived_by_sample from a sample that is neither a pool

Reply:
yeah its pretty much where derived sample name come from ... maybe just Derive_sample() you take a sample and you derive it into a new sample (and substructures)...